### PR TITLE
RouteDurations text centered in callout

### LINKED ISSUE #4080: [Bug]: RouteDurations Text Not Centering In Callout
### Mapbox Navigation SDK version

2.6.0

### Steps to reproduce

self!.mapView.showRouteDurations(along: response.routes!)

Produces text for route durations callouts that is not centered in the callout in V2 (See screenshot).  This wasn't an issue in V1 where the text was automatically centered in the callout and was appropriately sized (See screenshot). There is probably a way to manipulate the text in the callout, but I haven't been able to find it yet.  Any help you can give me is greatly appreciated.

V2_Callout_Text
V1_Callout_Text_Centered
 

### Expected behavior

Route Duration Text should fit the callout and be centered.

### Actual behavior

Route Duration Text is too small and is positioned at the bottom right of the callout.

### Is this a one-time issue or a repeatable issue?

repeatable

### PR DESCRIPTION
**Pull request by jill-cardamon**
### Description
This PR fixes #4080 by checking for a notch and adjusting the RouteDuration image and stretch accordingly.

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationMapView.swift [hunk 1]
```diff
@@ -1642,11 +1642,21 @@ open class NavigationMapView: UIView {
...
-            let stretchX = [ImageStretches(first: Float(33), second: Float(52))]
-            let stretchY = [ImageStretches(first: Float(32), second: Float(35))]
+            var stretchX: [ImageStretches]!
+            var stretchY: [ImageStretches]!
             // define the "content" area of the image which is the portion that the maps sdk will use
             // to place the text label within
-            let imageContent = ImageContent(left: 34, top: 32, right: 56, bottom: 50)
+            var imageContent: ImageContent!
+            
+            if !UIDevice.current.hasNotch {
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationMapView.swift:1651]: Would it make sense to replace `UIDevice.current.hasNotch` with actual top and bottom safe area inset values? They'll be 0 for devices without notch anyway.

File: Sources/MapboxNavigation/NavigationMapView+Annotations.swift [hunk 2]
```diff
@@ -35,15 +35,27 @@ extension NavigationMapView {
...
         
+        let bottomInset = UIApplication.shared.keyWindow?.safeAreaInsets.bottom ?? 0
+        
         // Right-hand pin
         if let image = Bundle.mapboxNavigation.image(named: "RouteInfoAnnotationRightHanded") {
             // define the "stretchable" areas in the image that will be fitted to the text label
             // These numbers are the pixel offsets into the PDF image asset
-            let stretchX = [ImageStretches(first: Float(33), second: Float(52))]
-            let stretchY = [ImageStretches(first: Float(32), second: Float(35))]
+            var stretchX: [ImageStretches]!
```
Comment (ShanMa1991) [inline Sources/MapboxNavigation/NavigationMapView+Annotations.swift:44]: The `stretchX` and `stretchY` don't change with the `bottomInset`. So we should keep them, and only update the `imageContent` with ternary operator based on the  `bottomInset`.  And the default `ImageContent(left: 28, top: 24, right: 58, bottom: 40)` is constant for two cases as well.

File: Sources/MapboxNavigation/NavigationMapView+Annotations.swift [hunk 3]
```diff
@@ -35,15 +35,17 @@ extension NavigationMapView {
...
         
+        let bottomInset = UIApplication.shared.keyWindow?.safeAreaInsets.bottom ?? 0
+        
         // Right-hand pin
         if let image = Bundle.mapboxNavigation.image(named: "RouteInfoAnnotationRightHanded") {
             // define the "stretchable" areas in the image that will be fitted to the text label
             // These numbers are the pixel offsets into the PDF image asset
-            let stretchX = [ImageStretches(first: Float(33), second: Float(52))]
-            let stretchY = [ImageStretches(first: Float(32), second: Float(35))]
+            var stretchX = [ImageStretches(first: Float(33), second: Float(52))]
```
Comment (ShanMa1991) [inline Sources/MapboxNavigation/NavigationMapView+Annotations.swift:44]: The `stretchX`, `stretchY` and `imageContent` don't change their values, we don't need to update them as variables.

File: Sources/MapboxNavigation/NavigationMapView+Annotations.swift [hunk 4]
```diff
@@ -30,60 +30,69 @@ extension NavigationMapView {
...
-                               content: imageContent)
-            
-            let selectedAnnotationImage = image.tint(routeDurationAnnotationSelectedColor)
-            try style.addImage(selectedAnnotationImage,
-                               id: "RouteInfoAnnotationLeftHanded-Selected",
-                               stretchX: stretchX,
-                               stretchY: stretchY,
-                               content: imageContent)
-        }
+        let capInsets = UIEdgeInsets(top: 15.0, left: 15.0, bottom: 35.0, right: 15.0)
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/NavigationMapView+Annotations.swift:48]: This approach allows to stop using magic numbers and make placement similar across different sceens (iPad, iPhone with/without notch etc). For example:

Simulator Screen Shot - iPad Air 2 - 2022-12-05 at 15 43 58

**Review by MaximAlien — APPROVED** [review]

**Pull request merged**